### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by albrow**
Still needs to be cleaned up a bit. Opening PR now so I can make sure it works in CI.

### CODE REVIEW

**Comment by albrow** [comment]
@fabioberger no longer WIP. This PR is ready for review.

File: Makefile [hunk 1]
```diff
@@ -42,6 +42,11 @@ test-go:
 	go test ./... -race -timeout 30s
 
 
+.PHONY: test-integration
+test-integration:
+	go test ./integration-tests -timeout 185s --integration
```
Comment (fabioberger) [inline Makefile]: 185s? How long do we expect them to take?

File: integration-tests/browser/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import { Mesh, OrderEvent, SignedOrder, BigNumber } from '@0x/mesh-browser';
+
+(async () => {
+    // Configure Mesh to use our local Ganache instance and network ID 51
```
Comment (fabioberger) [inline integration-tests/browser/src/index.ts]:
```suggestion
    // Configure Mesh to use our local Ganache instance and network ID 50
```

File: integration-tests/browser/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    mesh.onOrderEvents((events: Array<OrderEvent>) => {
+        for (let event of events) {
+            // console.log('received order event: ' + JSON.stringify(event));
+        }
+    });
+
+    // Start Mesh *after* we set up the handlers.
+    await mesh.startAsync();
+
+    // This order is for demonstration purposes only and is invalid. It will be
```
Comment (fabioberger) [inline integration-tests/browser/src/index.ts]: Please add to this comment _how_ it is invalid. Just nice to know.

File: Makefile (hunk 1 above)
Comment (fabioberger) [reply Makefile]: Ah yep, ran into that with webpack before. Really surprising it's not more optimized.

**Pull request merged**